power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -158,12 +158,39 @@ class Deserializer {
     std::vector<std::vector<ArraySpan>> msg_views_;
 
     static std::vector<char> json_to_msgpack(std::string_view json_string) {
-        nlohmann::json const json_document = nlohmann::json::parse(json_string);
+        nlohmann::json json_document = nlohmann::json::parse(json_string);
+        json_convert_inf(json_document);
         std::vector<char> msgpack_data;
         nlohmann::json::to_msgpack(json_document, msgpack_data);
         return msgpack_data;
     }
 
+    static void json_convert_inf(nlohmann::json& json_document) {
+        switch (json_document.type()) {
+        case nlohmann::json::value_t::object:
+        case nlohmann::json::value_t::array:
+            for (auto& value : json_document) {
+                json_convert_inf(value);
+            }
+            break;
+        case nlohmann::json::value_t::string:
+            json_string_to_inf(json_document);
+            break;
+        default:
+            break;
+        }
+    }
+
+    static void json_string_to_inf(nlohmann::json& value) {
+        std::string const str = value.get<std::string>();
+        if (str == "inf" || str == "+inf") {
+            value = std::numeric_limits<double>::infinity();
+        }
+        if (str == "-inf") {
+            value = -std::numeric_limits<double>::infinity();
+        }
```

Comment:
i wonder: is nlohmann json able to handle this?

i guess if the tests pass, then yes

Reply:
See the links below

https://json.nlohmann.me/features/types/number_handling/

https://json.nlohmann.me/features/binary_formats/messagepack/